cmd/kubeadm/app/util/config/strict/strict.go

```diff
@@ -19,41 +19,30 @@ package strict
 import (
 	"github.com/pkg/errors"
 
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/klog/v2"
-	"sigs.k8s.io/yaml"
-
-	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/scheme"
-	"k8s.io/kubernetes/cmd/kubeadm/app/componentconfigs"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
 )
 
-// VerifyUnmarshalStrict takes a YAML byte slice and a GroupVersionKind and verifies if the YAML
-// schema is known and if it unmarshals with strict mode.
-//
-// TODO(neolit123): The returned error here is currently ignored everywhere and a klog warning is thrown instead.
-// We don't want to turn this into an actual error yet. Eventually this can be controlled with an optional CLI flag.
```

Comment:
this todo isn't exactly resolved, but i don't think we can ever enable the strict errors as blocking.
at minimum the new warning for case sensitive fields has to run for a few more releases.

Reply:
personally, i'd expect the software to always error out on typos, mistakes in API fields, etc.
we could add a flag, but perhaps we can just add a feature gate and graduate it to alpha->beta->ga.
if we add the flag, keep the flag "false" by default and then switch it to "true", i wonder if there would be a use case for the "false" value. the feature gate at least has a clear timeline.

Reply:
A feature gate sounds like a good plan.